dev/README.md

```diff
@@ -0,0 +1,84 @@
+This directory contains docs and helper scripts for developing and debugging:
+
+- [`dev.sh`](./dev.sh) includes misc dev helpers
+- [`dev-features.sh`](./dev-features.sh) includes helpers for developing specific
+  nix-bitcoin features (like services)
+- [`topics`](./topics) features specific topics
+
+## Setup a nix-bitcoin dev env
+
+[`create-dev-env.sh`](./create-dev-env.sh) creates a directory with the following contents:
+- The nix-bitcoin source
+- A `bin` dir for helper scripts
+- A `scenarios.nix` file for custom test scenarios
+- A `.envrc` file that defines a [direnv](https://direnv.net/) environment,
+  mainly for adding nix-bitcoin and helper scripts to `PATH`
+
+### Installation
+
+1. [Install direnv](https://direnv.net/docs/installation.html).\
+   If you use NixOS (and Bash as the default shell), just add the following to your system config:
+   ```nix
+     environment.systemPackages = [ pkgs.direnv ];
+     programs.bash.interactiveShellInit = ''
```

Comment:
What's this for? In fedimint we just add [interactive bash to the dev shell to fix some issues with non-interactive bash](https://github.com/fedimint/fedimint/blob/4dca27643e673eba166d10ebe4cdbaa4f3fff888/flake.nix#L816)

Reply:
This installs direnv on a NixOS system.
direnv automatically sets up env vars when browsing to a project dir in a shell or when opening a project file in an editor (when editor support is installed).
The nix-bitcoin dev env exclusively uses direnv and no nix-shell.

Reply:
Oh, I see. I got confused.

BTW. `direnv` is nice and all and I do use it personally, but in projects using Nix, I generally ask people "to call `nix develop` and *optionally* setup direnv". Any advantage of hard direnv requirement?

Reply:
Yes:
- For our env, we just [set a few env vars](https://github.com/erikarvstedt/nix-bitcoin/blob/a7c31404ae4bfdb61a519ff9a73fcd0c800818f8/dev/dev-env/create.sh#L15) that simply point to already existing filesystem locations. No derivations need to be built, so we don't need any complex nix machinery.
- `nix develop` doesn't have access to the PWD, so for setting up repo-relative paths it would only work when called from the repo root. 
  This is bad UX. I simply want to pop up a shell anywhere in the repo and have my environment ready.
Actually, this can be worked around by using `shellHook`. In the nixbitcoin.org repo, which has an extensive `nix develop` shell, we [use this trick](https://github.com/fort-nix/nixbitcoin.org/blob/c2eb4fad9408a15520916dffb4d33ad2ed6843a2/lib/dev-env.nix#L21) to set env var `root` to point to the repo root, regardless of where `nix develop` is called.
Still, the problem that `nix develop` has to be called to enter the dev env remains.

Reply:
I've edited the previous post.

Reply:
I'll add `nix develop` support which just puts `helper` and `test/run-tests.sh` in PATH.

Reply:
> Users should be able to simply pop up a shell anywhere in the repo and have their environment ready.

But with direnv one still needs to `direnv allow` which is kind of the same as `nix develop`. And having to setup `direnv` is an extra step.

> For our env, we just [set a few env vars](https://github.com/erikarvstedt/nix-bitcoin/blob/a7c31404ae4bfdb61a519ff9a73fcd0c800818f8/dev/dev-env/create.sh#L15) that simply point to already existing filesystem locations.

`shellHook` can do that all the same, no?